src/domain/authentication/errors.ts

```diff
@@ -0,0 +1,3 @@
+import { DomainError } from "@domain/shared"
+
+export class AuthenticationError extends DomainError {}
```

Comment:
To follow with the patterns of the other subdomains I think we need an `AuthenticationServiceError`.